### ISSUE TITLE
[BUG] Switch to using ASyncTCP-esphome in v1.5 caused a crash in my code

### ISSUE THREAD

**Issue by judge2005** [bertmelis/espMqttClient#148]
#### Do not use to discuss topics! 

**Describe the bug**
My project had a dependency on v1.5 of this library and was initially working fine, however at some point you added a dependency to ASyncTCP-esphome and this resulted in an exception being thrown in ASyncWebServer (which was still the me-no-dev version). I'm sorry, I didn't save the stack trace, I was focussed on getting my device back and working at the time.

In the end I change the dependency on espMqttClient to explicitly load the v1.5.0 tag directly from github and my application worked fine again.

While the cause of the exception is not really your issue, your inclusion of a dependency on ASyncTCP-esphome creates a problem for me. Could you remove that dependency and just document that ASyncTCP-esphome is an alternate fork of ASyncTCP. It looks like it has been modified to work with LibreTiny, but the docs look to be just clones of the me-no-dev docs.

Again, I understand that issues that seem to be caused by ASyncTCP-esphome are not really your problem, but it would be helpful to remove the explicit dependency.

Platform is ESP32
Do you use TLS or not? No
Do you use an IDE (Platformio)?
Which version of the Arduino framework?

platform = espressif32 @ 6.5.0
board = esp32dev
framework = arduino
platform_packages = framework-arduinoespressif32 @ 3.20014.231204

**Comment by bertmelis** [comment]
So you're suggesting to remove the asynctcp dependency completely?
That might be the cleanest solution in the end...

**Comment by judge2005** [comment]
Right, I think that would be best.I did recently try the whole esphome asynchronous stack (TCP and Web), and then I don’t get the stack trace, but I do get a UTF-8 encoding error for web sockets that the me-no-dev version doesn’t show. I left them a bug report. OTH it has taken me a long time to get my software to be reliable over the long term so I am wary of jumping to a new version of a key library like that.Thanks for the reply.On Feb 25, 2024, at 8:26 AM, Bert Melis ***@***.***> wrote:﻿
So you're suggesting to remove the asynctcp dependency completely?
That might be the cleanest solution in the end...

**Comment by sfh-soem** [comment]
I have a similar problem to judge2005. I mainly use the following libraries in an ESP32 project:
- ESP Async WebServer v1.2.3 (Me-No-Dev)
- AsyncTCP v1.1.1 (Me-No-Dev)
- espMqttClient v1.5.0 (Bert Melis)

Everything works great. Since the update to espMqttClient v1.6.0 the system crashes when I access the web server. Probably there is a conflict with the additional AsyncTCP-esphome v2.1.3 library installed as a dependency with the new espMqttClient. I have solved the problem for me by continuing to use version 1.5.0. But I would prefer a more elegant solution.

Greetings, Scott-Falk Huehn

**Comment by bertmelis** [comment]
I removed the explicit dependencies.

Everyone is to choose the fork they like.
